libraries/AP_InertialSensor/AP_InertialSensor_Logging.cpp

```diff
@@ -98,12 +98,17 @@ void AP_InertialSensor::Write_Vibration() const
 // Write information about a series of IMU readings to log:
 bool AP_InertialSensor::BatchSampler::Write_ISBH(const float sample_rate_hz) const
 {
+    uint8_t instance_to_write = instance;
+    if (post_filter && (_doing_pre_post_filter_logging
+            || (_doing_post_filter_logging && _doing_sensor_rate_logging))) {
+        instance_to_write += (type == IMU_SENSOR_TYPE_ACCEL ? _imu._accel_count : _imu._gyro_count);
```

Comment:
i think we should reserve the first 2 bits of instance for IMU number (so it doesn't vary with number of IMUs), next 2 bits for filter stage (0==unfiltered, 1==first stage done, 2==2nd stage done)

Reply:
The reason I didn't do it this way is because it changes the output for *all* options rather than just having pre- and post-filtering. Currently if you don't use the option you get the same output. If we do it this way there is no backwards compatibility at all. So although if starting from scratch this might make sense, given where we are this is too much of a breaking change.